### ISSUE TITLE
gnome-shell journal error flood when dash-to-dock is viewing 

### ISSUE THREAD

**Issue by akrava** [micheleg/dash-to-dock#1332]
I've noticed, that extension is flooding too much in journalctl:
```
окт 19 23:41:58 UserPC gnome-shell[2975]: Object St.Bin (0x558698bafc10), has been already deallocated — impossible to set any property on it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
окт 19 23:41:58 UserPC gnome-shell[2975]: == Stack trace for context 0x55869671d900 ==
окт 19 23:41:58 UserPC gnome-shell[2975]: #0   7ffcec58aa60 b   /home/a_krava/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:2036 (1b57f85acf88 @ 194)
окт 19 23:41:58 UserPC gnome-shell[2975]: Object St.Bin (0x558698bafc10), has been already deallocated — impossible to set any property on it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
окт 19 23:41:58 UserPC gnome-shell[2975]: == Stack trace for context 0x55869671d900 ==
окт 19 23:41:58 UserPC gnome-shell[2975]: #0   7ffcec58aa60 b   /home/a_krava/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:2036 (1b57f85acf88 @ 194)
окт 19 23:41:58 UserPC gnome-shell[2975]: Object St.Bin (0x558698bafc10), has been already deallocated — impossible to set any property on it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
окт 19 23:41:58 UserPC gnome-shell[2975]: == Stack trace for context 0x55869671d900 ==
окт 19 23:41:58 UserPC gnome-shell[2975]: #0   7ffcec58aa60 b   /home/a_krava/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:2036 (1b57f85acf88 @ 194)
окт 19 23:41:58 UserPC gnome-shell[2975]: Object St.Bin (0x558698bafc10), has been already deallocated — impossible to set any property on it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
окт 19 23:41:58 UserPC gnome-shell[2975]: == Stack trace for context 0x55869671d900 ==
окт 19 23:41:58 UserPC gnome-shell[2975]: #0   7ffcec58aa60 b   /home/a_krava/.local/share/gnome-shell/extensions/dash-to-dock@micxgx.gmail.com/docking.js:2036 (1b57f85acf88 @ 194)
окт 19 23:41:58 UserPC gnome-shell[2975]: Object St.Bin (0x558698bafc10), has been already deallocated — impossible to set any property on it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose(), or remove() vfuncs.
```

The size of journal with such errors can be about 2-3 Gb. Can we fix it?
 
OS: Ubuntu 20.04.1 LTS x86_64 
Kernel: 5.4.0-51-generic 
Shell: bash 5.0.17 
DE: GNOME 
WM: Mutter 
Terminal: gnome-terminal  
Dash-to-dock version: 69
GNOME Shell 3.36.4

**Comment by brucehvn** [comment]
I'm seeing the same thing in syslog also:
Ubuntu 20.04.1 LTS
Kernel: 5.4.0-51-generic
Gnome Shell Version: 3.36.0-2ubuntu1
Dash to Dock Version: 69

```
Oct 20 07:19:59 BRUCE-CUSA gnome-shell[3133]: #0   7fff324a6220 b   /home/bruce/.local/share/gnome-shell/extensions/dash-to-dock@m
icxgx.gmail.com/docking.js:2036 (15d7267a3d30 @ 194)
Oct 20 07:19:59 BRUCE-CUSA gnome-shell[3133]: Object St.Bin (0x55965eabd470), has been already deallocated — impossible to set any
 property on it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose()
, or remove() vfuncs.
Oct 20 07:19:59 BRUCE-CUSA gnome-shell[3133]: == Stack trace for context 0x55965b3fc6e0 ==
Oct 20 07:19:59 BRUCE-CUSA gnome-shell[3133]: #0   7fff324a6220 b   /home/bruce/.local/share/gnome-shell/extensions/dash-to-dock@m
icxgx.gmail.com/docking.js:2036 (15d7267a3d30 @ 194)
Oct 20 07:19:59 BRUCE-CUSA gnome-shell[3133]: Object St.Bin (0x55965eabd470), has been already deallocated — impossible to set any
 property on it. This might be caused by the object having been destroyed from C code using something such as destroy(), dispose()
, or remove() vfuncs.
```

**Comment by vanvugt** [comment]
Downstream Ubuntu bug: https://bugs.launchpad.net/bugs/1905519

**Comment by upsuper** [comment]
Apparently this is caused by #1245. It seems that the urgent state is not cleared before the icon goes away, and thus `IconAnimator` keeps an invalid reference, which is used in a callback invoked on every new frame, thus this flood of error messages.

@rhendric @3v1n0 As the author and reviewer of the original PR, could you have a look at this issue?

**Comment by rhendric** [comment]
Just seeing this now. I'm attempting to reproduce.

**Comment by rhendric** [comment]
Can reproduce, working on a fix. Thanks @upsuper for bringing this to my attention.

**Comment by rhendric** [comment]
Anyone seeing this issue, could you apply #1366 and see if that resolves it? It works for me on GNOME Shell 3.38.2; I haven't tested on 3.36.

**Comment by upsuper** [comment]
I tested it on a Ubuntu 20.04.1 LTS virtual machine (which is using GNOME Shell 3.36.3), and it seems #1366 fixes the issue there. Thanks for the quick fix!

**Comment by upsuper** [comment]
For the record, my steps to reproduce / verify are:
1. open a terminal and run `sudo journalctl -f`
2. remove Rhythmbox from the dock
3. run Rhythmbox
4. execute the script below
5. wait until it prints "setting urgent flag"
6. close Rhythmbox
7. look at the first terminal

<details>
<summary>Test script</summary>

```python
import gi
gi.require_version('Unity', '7.0')

from gi.repository import Unity, Gio, GLib, GObject, Dbusmenu

loop = GLib.MainLoop()

# Pretend to be evolution for the sake of the example 
launcher = Unity.LauncherEntry.get_for_desktop_id ("rhythmbox.desktop")

# Show a count of 124 on the icon
launcher.set_property("count", 124)
launcher.set_property("count_visible", True)

# Set progress to 42% done 
launcher.set_property("progress", 0.42)
launcher.set_property("progress_visible", True)

# We also want a quicklist 
ql = Dbusmenu.Menuitem.new ()
item1 = Dbusmenu.Menuitem.new ()
item1.property_set (Dbusmenu.MENUITEM_PROP_LABEL, "Item 1")
item1.property_set_bool (Dbusmenu.MENUITEM_PROP_VISIBLE, True)
item2 = Dbusmenu.Menuitem.new ()
item2.property_set (Dbusmenu.MENUITEM_PROP_LABEL, "Item 2")
item2.property_set_bool (Dbusmenu.MENUITEM_PROP_VISIBLE, True)
ql.child_append (item1)
ql.child_append (item2)
launcher.set_property("quicklist", ql)

def update_urgency():
    if launcher.get_property("urgent"):
        print("Removing urgent flag")
        launcher.set_property("urgent", False)
    else:
        print("setting urgent flag")
        launcher.set_property("urgent", True)
    return True

GLib.timeout_add_seconds(5, update_urgency)

loop.run()
```
</details>

This script is largely based on the example on the [Ubuntu wiki](https://wiki.ubuntu.com/Unity/LauncherAPI#Python_Example) with some minor changes:
* `gi.require_version('Unity', '7.0')` is added before importing other things to fix a warning
* `GLib.MainLoop` and `GLib.timeout_add_seconds` are used instead of `GObject` as those on the latter are deprecated
* `rhythmbox` is picked over `evolution` because there is no `evolution.desktop` anymore (there is one `evolution-calendar.desktop`, though)
* use `print` function rather than statement

It seems I can't edit that page, so if anyone with the permission can help update the example script in that page, that would be appreciated.

**Comment by akrava** [comment]
@upsuper Thanks for investigation and involving contributors to fix this issue! I've reproduced this bug with your steps described in comment [above](https://github.com/micheleg/dash-to-dock/issues/1332#issuecomment-753229004), and after applying fix #1366 this issue is no more reproducing.
Thanks @rhendric, your fix is working, I've verified it on my machine. 
Thanks all you guys!

P.S. Happy New Year! :tada:
